packages/mdc-segmented-button/segment/_segment-theme.scss

```diff
@@ -0,0 +1,90 @@
+//
+// Copyright 2020 Google Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// Theme: Variables, functions, and mixins required to theme component segment
+
+@use "@material/feature-targeting/functions" as feature-targeting-functions;
+@use "@material/feature-targeting/mixins" as feature-targeting-mixins;
+@use "@material/theme/variables" as theme-variables;
+@use "@material/theme/custom-properties";
+
+$on-surface: theme-variables.prop-value(on-surface);
+$surface: theme-variables.prop-value(surface);
+$primary: theme-variables.prop-value(primary);
+
+$unselected-ink-color: custom-properties.create(
+  --mdc-segmented-button-unselected-ink-color,
+  rgba($on-surface, 0.6)
+);
```

Comment:
Can you define theme variables with only their respective values. You an use `custom-properties.create()` API inside the mixin where it was applied.

Example:

```scss
$unselected-ink-color: rgba($on-surface, 0.6);

@mixin ink-color($color) {
  $property: custom-properties.create(
    --mdc-segmented-button-unselected-ink-color,
    $color);
  );

  @include custom-property.apply('color', $property);
}

.mdc-segmented-button__segment {
  @include ink-color($unselected-ink-color);
}
```

This'll make sure that theme mixins (e.g., `ink-color()`) only accepts primitive types (i.e., `Color` or `Number`) excluding Sass map that contains arbitrary CSS custom property.

Reply:
This makes sense. I didn't really understand how custom-properties was supposed to be used.